lab/manifests/system/base/ironic/internal/base/deployment.yaml

```diff
@@ -59,37 +91,61 @@ spec:
               name: ironic-shared
             - mountPath: /etc/ironic
               name: ironic-conf
+
+        - name: init-credentials
+          image: alpine/k8s:1.27.2
+          command: [sh, -c]
+          args:
+            - |
+              apk add apache2-utils openssl
+
+              username=admin
+              password=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 64)
+              echo "$username" > /credentials/username
+              echo "$password" >  /credentials/password
+              htpasswd -b -c -B /credentials/htpasswd "$username" "$password"
+
+              kubectl create secret generic ironic-credentials --from-file=/credentials --dry-run=client -o yaml | kubectl apply -f -
+              kubectl delete pods -l app=baremetal-operator
```

Comment:
This looks unnecessary and ugly. Why to recreate the secret and restart bmo every time when ironic restarts? It should be ok to only create the secret if it doesn't exist.